### PR TITLE
Fix/message passing typo and docs

### LINKED ISSUE #1757: FactorGraph message passing
### Subject of the issue
The are various things that I was wondering about when starting to use the message passing algorighm for FactorGraph

- [ ] BeliefPropagationWithMessageParsing shoud be called ...MessagePassing
- [ ] is not listed in the doc https://pgmpy.org/exact_infer/bp.html

### Your environment
* 0.1.25

### PR DESCRIPTION
**Pull request by tillwenke**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes #1757

### List of changes to the codebase in this pull request
- rename all occurences of BeliefPropagationWithMessageParsing to BeliefPropagationWithMessagePassing
- add BeliefPropagationWithMessagePassing class to the docs

### CODE REVIEW

File: docs/exact_infer/bp_wmp.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Belief Propagation with Message Passing
+==================
```
Comment (ankurankan) [inline docs/exact_infer/bp_wmp.rst]: @tillwenke I think the `====` line needs to be at least as long as the text: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html

**Comment by ankurankan** [comment]
@tillwenke Thanks for creating the PR. I have added a minor comment on the formatting for the docs.

File: docs/exact_infer/bp_wmp.rst (hunk 1 above)
Comment (tillwenke) [reply docs/exact_infer/bp_wmp.rst]: done

**Comment by ankurankan** [comment]
@tillwenke Thanks for the PR

**Pull request merged**